octopoes/octopoes/xtdb/query.py

```diff
@@ -146,17 +167,19 @@ def _where_field_is(self, ref: Ref, field_name: str, value: Ref | str | set[str]
 
         abstract_types = get_abstract_types()
 
+        if isinstance(value, str):
+            value = value.replace('"', r"\"")
+
         if ooi_type in abstract_types:
             if isinstance(value, str):
-                value = value.replace('"', r"\"")
-                self._add_or_statement(ref, field_name, f'"{value}"')
+                self._add_or_statement_for_abstract_types(ref, field_name, f'"{value}"')
                 return
 
             if not isinstance(value, type):
```

Comment:
I added a test that does some checking, but it took a while to realize that this check is needed because `issubclass` fails if the first argument is not a class:

We need the subclass check to verify if the type is an OOI Type:
```python
>>> value = Network
>>> isinstance(value, OOIType)
False
>>> isinstance(value, OOI)
False
>>> isinstance(ooi_type, type)
True
>>> issubclass(ooi_type, OOI)
True
>>> issubclass(3, OOI)
Traceback (most recent call last):
  [...]
TypeError: issubclass() arg 1 must be a class
```

The test also spotted a small bug :relieved:

Reply:
`isinstance` only works for instances, so you'd have to give it `Network()` rather than type `Network`. `isinstance(value_type, type)` will always return `True` since everything derives from class `type`. And `issubclass` works for class types and not instances